src/lib/structures/expr.ml

```diff
@@ -1438,10 +1441,14 @@ and mk_forall_bis (q : quantified) =
         let q = {q with binders; user_trs = trs; sko_v; main = f } in
         mk_forall_bis q
 
+(* If [f] is a formula of the form [x = a -> P(x)] where [a] doesn't content
+   [x], this function produces the substitution { x |-> a }.
+
+   *Note*: this heuristic isn't used if the user has defined filters.
+
+   @return [None] if the formula hasn't the right form. *)
```

Comment:
I think keeping the example as in your original rewriting was useful (or at least mention that we work with CNF so that `x = a -> P(x)` is written `x <> a \/ P(x)`).